pkg/cloudclient/aws/private.go

```diff
@@ -209,34 +209,39 @@ func (c *Client) validateEgress(ctx context.Context, VPCSubnetID string) error {
 	// Generate the userData file
 	userData, err := generateUserData()
 	if err != nil {
-		panic(fmt.Sprintf("Unable to generate UserData file: %s\n", err.Error()))
+		fmt.Printf("Unable to generate UserData file: %s\n", err.Error())
+		return err
 	}
 
 	// Create an ec2 instance
 	instance, err := createEC2Instance(c.ec2Client, AMIID, InstanceType, InstanceCount, VPCSubnetID, userData)
 	if err != nil {
-		panic(fmt.Sprintf("Unable to create EC2 Instance: %s\n", err.Error()))
+		fmt.Printf("Unable to create EC2 Instance: %s\n", err.Error())
+		return err
 	}
 	instanceID := *instance.Instances[0].InstanceId
 
 	// Wait for the ec2 instance to be running
 	fmt.Printf("Waiting for EC2 instance %s to be running\n", instanceID)
 	err = waitForEC2InstanceCompletion(c.ec2Client, instanceID)
 	if err != nil {
-		panic(err)
+		fmt.Printf("Error while waiting for EC2 instnace to start: %s\n", err.Error())
+		return err
 	}
 	fmt.Println("Gather and parse console log output")
 	unreachableEndpoints, err := findUnreachableEndpoints(c.ec2Client, instanceID)
 	if err != nil {
-		panic(err)
+		fmt.Printf("Error parsing output from console log: %s\n", err.Error())
+		return err
 	}
 
 	fmt.Println("Terminating instance")
 	if err := terminateEC2Instance(c.ec2Client, instanceID); err != nil {
-		panic(err)
+		fmt.Printf("Error terminating instances: %s\n", err.Error())
+		return err
 	}
 	if len(unreachableEndpoints) > 0 {
-		return errors.New(fmt.Sprintf("multiple target unreachable %q", unreachableEndpoints))
+		return fmt.Errorf("multiple target unreachable %q", unreachableEndpoints)
```

Comment:
nitpick: s/target/targets/

Another nitpick: Inconsistent capitalization of errors messages throughout.

Not worth blocking the MR.

Reply:
Errorf gives a warning that says not to capitalize, while Printf doesn't :(